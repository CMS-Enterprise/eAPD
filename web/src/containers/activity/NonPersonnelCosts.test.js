import { shallow } from 'enzyme';
import React from 'react';

import {
  plain as NonPersonnelCosts,
  mapStateToProps,
  mapDispatchToProps
} from './NonPersonnelCosts';

import {
  addNonPersonnelCost,
  removeNonPersonnelCost
} from '../../actions/editActivity';

describe('activity non-personnel costs subsection', () => {
  const props = {
    activityIndex: 58,
    addExpense: jest.fn(),
    expenses: [
      {
        category: 'test category',
        desc: 'test desc',
        key: 'cost key',
        years: {
          2027: 34355,
          2028: 48833
        }
      }
    ],
    removeExpense: jest.fn()
  };
  const component = shallow(<NonPersonnelCosts {...props} />);

  beforeEach(() => {
    props.addExpense.mockClear();
    props.removeExpense.mockClear();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  describe('events', () => {
    const list = component.find('FormAndReviewList');

    it('handles adding a new cost', () => {
      list.prop('onAddClick')();
      expect(props.addExpense).toHaveBeenCalledWith(58);
    });

    it('handles deleting a cost', () => {
      list.prop('onDeleteClick')('cost key');
      expect(props.removeExpense).toHaveBeenCalledWith(58, 0);
    });
  });

  it('maps state to props', () => {
    expect(
      mapStateToProps(
        {
          apd: {
            data: {
              activities: [
                {
                  expenses: 'these are wrong expenses'
                },
                {
                  expenses: 'these are wrong expenses'
                },
                {
                  expenses: 'these are expenses'
                }
              ]
            }
          }
        },
        { activityIndex: 2 }
      )
    ).toEqual({ expenses: 'these are expenses' });
  });

  it('maps dispatch actions to props', () => {
    expect(mapDispatchToProps).toEqual({
      addExpense: addNonPersonnelCost,
      removeExpense: removeNonPersonnelCost
    });
  });
});
