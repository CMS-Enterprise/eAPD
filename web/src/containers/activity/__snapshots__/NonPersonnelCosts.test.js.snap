// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity non-personnel costs subsection renders correctly 1`] = `
<Fragment>
  <Instruction
    args={null}
    headingDisplay={
      Object {
        "className": "ds-h3",
        "level": "h3",
      }
    }
    reverse={false}
    source="activities.expenses.instruction"
  />
  <FormAndReviewList
    activityIndex={58}
    addButtonText="Add another non-personnel cost"
    allowDeleteAll={false}
    className={null}
    collapsed={[Function]}
    expanded={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(NonPersonnelCostForm)",
        "type": [Function],
      }
    }
    list={
      Array [
        Object {
          "category": "test category",
          "desc": "test desc",
          "initialCollapsed": false,
          "key": "cost key",
          "years": Object {
            "2027": 34355,
            "2028": 48833,
          },
        },
      ]
    }
    noDataMessage="Additional costs for this activity are missing."
    onAddClick={[Function]}
    onDeleteClick={[Function]}
  />
</Fragment>
`;
